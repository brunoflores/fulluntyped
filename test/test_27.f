-- Streams.

Stream = Rec A. Unit -> { Nat, A };

hd = lambda s: Stream. (s unit).1;
tl = lambda s: Stream. ((s unit).2) as Stream;

upfrom0 = fix (lambda f: Nat -> Stream. lambda n: Nat. lambda _: Unit. { n, f (succ n) }) 0;

hd upfrom0;
hd (tl (tl (tl upfrom0)));
