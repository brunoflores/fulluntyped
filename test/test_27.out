Stream: Rec A. Unit -> { Nat, A }
hd: Stream -> Nat
tl: Stream -> Stream
upfrom0: Unit -> { Nat, Stream }
0: Nat
3: Nat
